odoo/tests/common.py

```diff
@@ -2111,10 +2111,14 @@ def _process_fvg(self, model, fvg, level=2):
             fname = f.get('name')
             order.append(fname)
 
-            node_modifiers = {
-                modifier: ([TRUE_LEAF] if domain else [FALSE_LEAF]) if isinstance(domain, int) else normalize_domain(domain)
-                for modifier, domain in json.loads(f.get('modifiers', '{}')).items()
-            }
+            node_modifiers = {}
+            for modifier, domain in json.loads(f.get('modifiers', '{}')).items():
+                if isinstance(domain, int):
+                    node_modifiers[modifier] = [TRUE_LEAF] if domain else [FALSE_LEAF]
+                elif isinstance(domain, str):
+                    node_modifiers[modifier] = normalize_domain(safe_eval(domain, dict(self._env.context), nocopy=True))
```

Comment:
Why `nocopy=True`?  Is `safe_eval` expected to have side effects on the evaluation context?